lib/ansible/modules/network/fortimanager/fmgr_secprof_dns.py

```diff
@@ -0,0 +1,483 @@
+---
+module: fmgr_secprof_dns
+version_added: "2.8"
+author:
+    - Luke Weighall (@lweighall)
+    - Andrew Welsh (@Ghilli3)
+    - Jim Huber (@p4r4n0y1ng)
+short_description: Manage DNS security profiles in FortiManager
+description:
+  -  Manage DNS security profiles in FortiManager
+
+options:
+  adom:
+    description:
+      - The ADOM the configuration should belong to.
+    required: false
+    default: root
+
+  host:
+    description:
+      - The FortiManager's Address.
+    required: true
+
+  username:
+    description:
+      - The username associated with the account.
+    required: true
+
+  password:
+    description:
+      - The password associated with the username account.
+    required: true
+
+  mode:
+    description:
+      - Sets one of three modes for managing the object.
+      - Allows use of soft-adds instead of overwriting existing values
+    choices: ['add', 'set', 'delete', 'update']
+    required: false
+    default: add
+
+  youtube_restrict:
+    type: str
+    description:
+      - Set safe search for YouTube restriction level.
+      - choice | strict | Enable strict safe seach for YouTube.
+      - choice | moderate | Enable moderate safe search for YouTube.
+    required: false
+    choices: ["strict", "moderate"]
+
+  sdns_ftgd_err_log:
+    type: str
+    description:
+      - Enable/disable FortiGuard SDNS rating error logging.
+      - choice | disable | Disable FortiGuard SDNS rating error logging.
+      - choice | enable | Enable FortiGuard SDNS rating error logging.
+    required: false
+    choices: ["disable", "enable"]
+
+  sdns_domain_log:
+    type: str
+    description:
+      - Enable/disable domain filtering and botnet domain logging.
+      - choice | disable | Disable domain filtering and botnet domain logging.
+      - choice | enable | Enable domain filtering and botnet domain logging.
+    required: false
+    choices: ["disable", "enable"]
+
+  safe_search:
+    type: str
+    description:
+      - Enable/disable Google, Bing, and YouTube safe search.
+      - choice | disable | Disable Google, Bing, and YouTube safe search.
+      - choice | enable | Enable Google, Bing, and YouTube safe search.
+    required: false
+    choices: ["disable", "enable"]
+
+  redirect_portal:
+    type: str
+    description:
+      - IP address of the SDNS redirect portal.
+    required: false
+
+  name:
+    type: str
+    description:
+      - Profile name.
+    required: false
+
+  log_all_domain:
+    type: str
+    description:
+      - Enable/disable logging of all domains visited (detailed DNS logging).
+      - choice | disable | Disable logging of all domains visited.
+      - choice | enable | Enable logging of all domains visited.
+    required: false
+    choices: ["disable", "enable"]
+
+  external_ip_blocklist:
+    type: str
+    description:
+      - One or more external IP block lists.
+    required: false
+
+  comment:
+    type: str
+    description:
+      - Comment.
```

Comment:
Any chance you can expand on this? If a user puts in a comment, where does it show up? why would I want to add a comment? Does it go in a log somewhere for example? show up on an output to help explain 'something'?